northstar/src/runtime/process/mod.rs

```diff
@@ -278,27 +299,45 @@ fn waitpid(
     pid: Pid,
     tx: EventTx,
     stop: CancellationToken,
+    mut channel: pipe::AsyncChannelRead,
 ) -> impl Future<Output = ExitStatus> {
-    let mut sigchld = signal::unix::signal(signal::unix::SignalKind::child())
-        .expect("Failed to set up signal handle for SIGCHLD");
-
     task::spawn(async move {
-        // Check the status of the process after every SIGCHLD is received
-        let exit_status = loop {
-            sigchld.recv().await;
-            if let Some(exit) = exit_status(pid) {
-                stop.cancel();
-                break exit;
+        let exit_status = match channel.recv::<ExitStatus>().await {
+            Ok(exit_status) => {
+                debug!(
+                    "Received exit status of {} via channel: {}",
+                    container, exit_status
+                );
+
+                // Wait for init to exit
+                let pid = unistd::Pid::from_raw(pid as i32);
+                wait::waitpid(Some(pid), None).ok();
```

Comment:
The `waitpid` is  needed because you can have the situation that the `init` process exits and within the exit it closes the pipe. The runtime starts to process this exit by tearing down cgroups but the process is not yet gone. Therefore wait until the pipe is closed *and* the process is completely exited.